src/vaadin-crud.html

```diff
@@ -813,6 +815,15 @@ <h3 slot="header">[[__computeEditorHeader(__isNew, i18n.newItem, i18n.editItem)]
           }
         }
 
+        /** @private */
+        __cancelByEscOrClick(e) {
+          if (e.type == 'vaadin-overlay-escape-press' && !this.__isDirty ||
+              e.type == 'vaadin-overlay-outside-click' && !this.__isDirty) {
+            e.preventDefault();
+          }
```

Comment:
I think this logic is unnecessary (the lines can be removed and all the tests still pass), since the dialog already gets `noCloseOnOutsideClick` and `noCloseOnEsc` applied whenever `this.__isDirty` [here](https://github.com/vaadin/vaadin-crud/blob/master/src/vaadin-crud.html#L75).

If that's the case, the `__cancelByEscOrClick` function shouldn't be needed since the events could be bound to `this. __cancel` directly.

Reply:
Yes, it looks like so, I was just having play it safe approach.